### ISSUE TITLE
Modernizr firing off Complete callback for magnific-popup before it's loaded.

### ISSUE THREAD

**Issue by CalvinRodo** [wet-boew/wet-boew#8083]
We are running into an issue in IE (repeatable on IE11) where modernizr is firing off the `complete` callback for the magnific-popup error before the script is being loaded, so we receive the following error: 

```
SCRIPT5007: Unable to get property 'defaults' of undefined or null reference
File: wet-boew.min.js, Line: 8, Column: 70652
```

This seems to be related to #7620

You can see more information here [Intranet GC/Code Link](http://gitlab.ssc.etg.gc.ca/GOCWebTemplates/DotNetTemplates/issues/108)

**Comment by LaurentGoderre** [comment]
The complete event means that it's loaded in memory, not that it has finished running. This is one of the biggest weakness of WET.

**Comment by CalvinRodo** [comment]
So since it's a known issue does this mean it's not going to be fixed anytime soon?

**Comment by CalvinRodo** [comment]
So I did some research with the DotNetWebTemplates http://gitlab.ssc.etg.gc.ca/GOCWebTemplates/DotNetTemplates v4.0.24 does not attempt to load jquery.magnific-popup.min.js whereas v4.0.25 does load that plugin.

**Comment by CalvinRodo** [comment]
Is it possible to wait in the complete callback for processing to finish in the plugin that's being loaded?

**Comment by LaurentGoderre** [comment]
That's not a small change. I had done it in my prototype for WET 5 but would require a lot of work to do that in WET4.

**Comment by CalvinRodo** [comment]
It's a bit of an issue for some of our developers so it would be nice to have it fixed. 

I would have imagined it would just be throwing some code into the Complete callback to wait for `$.magnificPopup !== null`

**Comment by LaurentGoderre** [comment]
yep but it's not just that plugin that is affected.

**Comment by CalvinRodo** [comment]
well we don't have that many plugins surely it can't be that big of a deal to add a check into the complete callbacks for them

**Comment by NoisyMonk** [comment]
I don't suppose there's any update to this? I'm having the same problem.

**Comment by stego39** [comment]
This is causing us issues as well... in that we rely on form validation and session timeout widgets in our app.. but whenever this issue occurs, it'll break other plugins and/or widgets randomly.. depending on what has or hasn't loaded yet.

Seems there is no fix in sight yet for this however.

**Comment by LaurentGoderre** [comment]
There is a fix. @StdGit fixed one of them in #8291. Any PR to bring this to other dependencies are welcome!

**Comment by StdGit** [comment]
@NoisyMonk and @stego39

FYI When the magnific-popup (Which is used in the session timeout) wasn't loaded, it broke jquery and the following scripts.

The fix intends to wait for the completion of the the magnific-popup before loading additional stuff. I think we have this bug lately because the network is really slow. 

If you think that your bug is caused by another dependency, let me know.

**Comment by RobJohnston** [comment]
Upgrading to 4.0.28.1 resolved this problem for me, likely due to the fix in #8291.

![image](https://user-images.githubusercontent.com/1347487/42323797-627e7a9c-802f-11e8-94de-fab1c47b3497.png)

**Comment by StdGit** [comment]
@RobJohnston 
Glad to hear it

**Comment by shawnthompson** [comment]
@CalvinRodo can you confirm this is fixed and can be closed?

**Comment by duboisp** [comment]
Closing this issue as it was fixed via PR #8291
